src/nunavut/lang/cpp/support/variable_length_array.hpp

```diff
@@ -355,6 +640,65 @@ class VariableLengthArray
     }
 
 private:
+    template <typename U>
+    static constexpr bool
+        internal_compare_element(const U& lhs, const U& rhs, typename std::enable_if<!std::is_floating_point<U>::value>::type* = 0) noexcept
+    {
+        return (lhs == rhs);
+    }
+
+    template <typename U>
+    static constexpr bool
+        internal_compare_element(const U& lhs, const U& rhs, typename std::enable_if<std::is_floating_point<U>::value>::type* = 0) noexcept
+    {
+        // From the C++ documentation for std::numeric_limits<T>::epsilon()
```

Comment:
I'm less sure about this choice. I needed to provide some sort of floating-point comparison specialization or the container would not support comparisons (I think std::variant requires this support?). My implementation is correct from an IEEE-754 standpoint (I think - this is why you are reviewing) but it is not correct if the units represented by different floating point data is different.

I wonder if we need an additional template parameter for the VLA type that represents units (ULP, perhaps?) such that two VLAs that happened to be the same floating point type and the same max size could not be compared using the built-in comparator?

Reply:
>I think std::variant requires this support?

You can instantiate `std::variant<Ts...>` even for Ts that do not define comparison operators. Given that, do we really want to define an opinionated default comparison for VLA?

>I wonder if we need an additional template parameter for the VLA type that represents units (ULP, perhaps?)

Why not simply allow the user to define a free-standing spaceship operator arbitrarily?

Reply:
I went back and forth on this. Without this support no generated C++ type that uses or depends on a VLA type can be compared. Is that acceptable? I'm not quite sure.